docs/DeveloperGuide.md

```diff
@@ -380,6 +380,59 @@ It is designed to preserve the Command Design Pattern. Through the implementatio
     * Cons:
         * No exact knowledge whether the execution of command mark task as complete or incomplete
 
+### Sorting
+
+#### What is the feature about
+Sorts the tasks presented by specified attribute and order.
+
+#### How the feature is implemented
+The sort feature uses the `sort` command, prefix `by/` before the specified `SORT_KEY` and prefix `in/` before the specified `SORT_ORDER`.
+
+Given below is an example usage scenario of how the sort mechanism behaves at each step to sort the task:
+
+Step 1. User inputs `sort by/deadline in/asc` to sort the task list by the `Deadline` in the ascending order.
+    
+Step 2. Upon receiving the user's input, `LogicManager` calls `HarmoniaParser#parseCommand()` to parse the user input.
+
+Step 3. The first word of the user input is `sort`, which matches the command for `SortCommand`. This initialises `SortCommandParser`.
+
+Step 4. `SortCommandParser#parse()` is called. `SORT_KEY` with prefix `by/` and `SORT_ORDER` with prefix `in/` are extracted out as a `SortKey` and `SortOrder` objects.
+
+Step 5. A `SortCommand` is initialised using the `SortKey` and `SortOrder`. The `SortCommand` constructor creates a `Comparator` used for sorting the list. `SortCommand` is returned to `LogicManager` for execution.
+
+Step 6. After `SortCommand#execute()` is called, `model#updateSortedTaskList(Comparator)` is invoked to sort the task list using the created `Comparator`. The command result is returned and displayed to the user.
+
+#### Why it is implemented that way
+The underlying `UniqueTaskList` uses an `ObservableList` and therefore it allows the application listen to changes and render according specified requirements. One such abstraction used in the initial implementation is the `FilteredList`. With the `FilteredList` the application grants the ability to filter the list using a `Predicate`. Sorting was designed to be an extension of this concept using `SortedList` abstraction where the application can sort the list by a given `Comparator` and render its output instantaneously.
+
+#### Design considerations:
+**Aspect: How the `Comparator` is created:**
+* **Alternative 1 (current choice):** Use Factory design pattern.
+    * Pros:
```

Comment:
Just a minor suggestion that you can consider if you have time! Would a possible pro be about how it promotes [loose-coupling](https://www.javatpoint.com/factory-method-design-pattern), among some other benefits such as [reuse and improving testability](https://softwareengineering.stackexchange.com/questions/200647/why-is-the-factory-method-design-pattern-more-useful-than-having-classes-and-cal)?